streaming_timeseries/streaming_timeseries.ipynb

```diff
@@ -0,0 +1,948 @@
+    "            )\n",
+    "            return plot\n",
+    "        return self.bare_pos_plot()\n",
+    "\n",
+    "    def bare_stream_plot(self, min_height: int = 600) -> hv.Curve:\n",
+    "        curve = hv.Curve([]).opts(\n",
+    "            yaxis=\"bare\",\n",
+    "            xaxis=\"bare\",\n",
+    "            min_height=min_height,\n",
+    "            responsive=True,\n",
+    "        )\n",
+    "        return curve\n",
+    "\n",
+    "    def empty_stream_plot(self) -> hv.NdOverlay:\n",
+    "        empty_curves = {\n",
+    "            ch: hv.Curve([]).relabel(ch).opts(\n",
+    "                subcoordinate_y=True,\n",
+    "                color=self.color_mapping.get(ch, None),\n",
+    "            )\n",
+    "            for ch in self.channel_names\n",
+    "        }\n",
+    "        ndoverlay = hv.NdOverlay(empty_curves).opts(\n",
+    "            legend_position=\"right\",\n",
+    "            responsive=True,\n",
+    "            min_height=600,\n",
+    "            title=\"\",\n",
+    "            show_legend=False,\n",
+    "        )\n",
+    "        return ndoverlay\n",
+    "\n",
+    "    def bare_pos_plot(self) -> hv.Points:\n",
+    "        points = hv.Points([]).opts(\n",
+    "            xaxis=None,\n",
+    "            yaxis=None,\n",
+    "            axiswise=True,\n",
+    "            height=300,\n",
+    "            responsive=True,\n",
+    "            shared_axes=False,\n",
+    "            title=\"Channel Position\",\n",
+    "        )\n",
+    "        return points\n",
+    "\n",
+    "    def __panel__(self) -> pn.Row | pn.Column:\n",
+    "        if self.notebook:\n",
+    "            return self.layout\n",
+    "        else:\n",
+    "            return self.template"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4915afc7-61b4-45a6-ae07-30e40d5a676b",
+   "metadata": {},
+   "source": [
+    "### Running the Application\n",
+    "\n",
+    "Instantiate the data sources and create the application."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9455b513-fc28-479d-b784-6be45c1dc693",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data_source_cpu_usage = CPU_Usage()\n",
+    "\n",
+    "# display a subset of eeg channels with 'picks'\n",
+    "picks = [\"F4\", \"F7\", \"FC5\", \"FC2\", \"T7\", \"CP2\", \"CP6\", \"P4\", \"P7\", \"O1\", \"O2\"]\n",
+    "data_source_eeg_usage = LSL_EEG_File_Stream(\n",
+    "    \"data/sample-ant-raw.fif\",\n",
+    "    picks=picks,\n",
+    ")\n",
+    "\n",
+    "nb_app = StreamingApp(data_sources=[data_source_eeg_usage, data_source_cpu_usage])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d48d63d4-c1ab-4102-bad2-20b85cbc3b0b",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "nb_app"
```

Comment:
it might be cool if we could have it trigger the cpu stream for a bit but only in the website build and otherwise hide the code from the tutorial notebook. But I don't have time to figure that out and also idk what it might even show for CPU usage on a CI system. So, dropping given urgency of initial release.. maybe we can come back to this later

Reply:
Just realized that the `await` here works fine in the context of a notebook runtime (where there's already an event loop), but I'm not sure Python would even let the app being served (SyntaxError?).

An other approach inspired by the simpler example Demetris added would be to instantiate the class with a `timeout` and pass it down to the callback.